test/unit/test_ext_auth.py

```diff
@@ -0,0 +1,262 @@
+import json
+from datetime import datetime
+
+import pytest
+
+from assertpy import assert_that
+from pcluster_dcv_ext_auth.pcluster_dcv_ext_auth import (
+    DCVAuthenticator,
+    OneTimeTokenHandler,
+    generate_random_token,
+    generate_sha512_hash,
+)
+
+MAIN = "pcluster_dcv_ext_auth.pcluster_dcv_ext_auth."
+EXT_AUTH = MAIN + "DCVAuthenticator."
+
+
+class TestOneTimeTokenHandler:
+    def __init__(self):
+        TestOneTimeTokenHandler.test_fixed_length()
+        TestOneTimeTokenHandler.test_correct_storage()
+        TestOneTimeTokenHandler.test_token_remove()
+
+    @staticmethod
+    def test_fixed_length():
+        storage = OneTimeTokenHandler(3)
+        storage.add_token("caso", ("patata", 1, 15.2, ["a", 2]))
```

Comment:
I'd use better names for the tested variable, like "wrong_token", "invalid_user", "invalid_token", "test_token" etc

Reply:
Done in the new patch.